core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -225,4 +249,44 @@ private void mockUserGroupInformation() throws IOException {
     final UserGroupInformation ugi = Mockito.mock(UserGroupInformation.class);
     Mockito.when(UserGroupInformation.getCurrentUser()).thenReturn(ugi);
   }
+
+  /**
+   * Sets returned status when listStatus is called in Alluxio FS.
+   *
+   * @param path the path to list status
+   * @param infos the list of file infos
+   */
+  private void mockAlluxioFsListStatusHelper(Path path, FileInfo... infos) throws Exception {
+    List<URIStatus> uris = new ArrayList<>();
+    for (FileInfo info : infos) {
+      uris.add(new URIStatus(info));
+    }
+    alluxio.client.file.FileSystem alluxioFs =
+        Mockito.mock(alluxio.client.file.FileSystem.class);
+    Whitebox.setInternalState(FileSystem.class, "sFileSystem", alluxioFs);
```

Comment:
@aaudiber actually that will not work. The reason is the instance `alluxio.hadoop.FileSystem` is created from Hadoop code by reflection with no constructor:
```java
URI uri = URI.create(Constants.HEADER + "localhost:19998/");
    org.apache.hadoop.fs.FileSystem alluxioHadoopFs =
        org.apache.hadoop.fs.FileSystem.get(uri, getConf());
```
We could directly create an `alluxio.hadoop.FileSystem` instance rather from the Hadoop factory code, but that will cover less.